portfolio/src/main/java/com/google/sps/servlets/FactsServlet.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns fun facts. */
+@WebServlet("/facts")
+public class FactsServlet extends HttpServlet {
+
+  /* List of fun facts */
+  private List<String> facts;
+  /* Pointer to the current index of facts given when the servlet is called */
+  private int currQuestionIndex;
+
+  /* Initializes facts list and currQuestionIndex */
+  @Override
+  public void init() {
+    currQuestionIndex = 0;
+    facts = new ArrayList<>();
```

Comment:
I'm not sure how to shuffle the array efficiently without using an ArrayList? I think that the space efficiency shouldn't be _too_ different if I switched to a primitive array and I'm only doing array accesses for the most part which is constant-time for ArrayLists, so I think it should be fine?

Reply:
It can be accomplished with a primitive array and the Arrays.asList function: https://www.journaldev.com/32661/shuffle-array-java

This is a minor improvement, though. I will approve this change regardless, but updating this code may be nice if it's not overwritten later by another change.